### PR TITLE
use third party prompt

### PR DESCRIPTION
**Pull request by charlierudolph**
It has select and multi-select (which we can use for choosing perennial branches). https://github.com/AlecAivazis/survey

Changes
- Can no longer enter a custom author (I've never used this anyway)
- Squash commit author list no longer lists how many commits each author has but the authors are still sorted in this manner. I'll open an issue on the lib to see if they'd support adding extra info to each option but I don't think thats a very important feature.
- Updated perennial branch options to not include the main branch (in place of validating main was not selected)
- Updated parent branch options to not include self / child branches (in place of detecting loops)

### CODE REVIEW

File: features/git-town-new-pull-request/github.feature [hunk 1]
```diff
@@ -72,21 +72,3 @@ Feature: git-new-pull-request when origin is on GitHub
...
-    Then I see a new GitHub pull request for the "feature" branch in the "Originate/git-town" repo in my browser
-
-
-  Scenario: nested feature branch with unknown parent (accepting default choice)
-    Given my repository has a feature branch named "feature"
-    And Git Town has no branch hierarchy information for "feature"
-    And my repo's remote origin is git@github.com:Originate/git-town.git
-    And I am on the "feature" branch
-    When I run `git-town new-pull-request` and press ENTER
-    Then I see a new GitHub pull request for the "feature" branch in the "Originate/git-town" repo in my browser
```
Comment (charlierudolph) [inline features/git-town-new-pull-request/github.feature]: Moved one of these to `features/shared/prompting_for_parent_branch.feature`

File: glide.lock [hunk 2]
```diff
@@ -23,6 +23,8 @@ imports:
 - name: github.com/mattn/go-isatty
   version: 57fdcb988a5c543893cc61bce354a6e24ab70022
   repo: https://github.com/mattn/go-isatty
+- name: github.com/mgutz/ansi
+  version: 9520e82c474b0a04dd04f8a40959027271bab992
```
Comment (charlierudolph) [inline glide.lock]: Glide didn't install this package until I started used it (a dependency of survey) until I started using it

File: features/git-town-config/setup.feature [hunk 3]
```diff
@@ -11,83 +11,9 @@ Feature: Initial configuration
...
 
   Scenario: succeeds on valid main branch and perennial branch names
-    When I run `git-town config --setup` and enter:
-      | main       |
-      | production |
-      | dev        |
-      |            |
+    When I run `git-town config --setup` and answer the prompts:
+      | PROMPT                                     | ANSWER                    |
+      | Please specify the main development branch | DOWN, ENTER               |
```
Comment (kevgo) [inline features/git-town-config/setup.feature]: What do you think about distinguishing pressed keys better from text that is entered? What I often see is something like this:

```
[DOWN] [ENTER]
```

Alternatively, we could use unicode:

```
⬇️ ⏎
```

Comment (charlierudolph) [reply features/git-town-config/setup.feature]: Updated to `[DOWN][ENTER]`

**Pull request merged**